server/src/main/java/org/opensearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -257,9 +272,77 @@ public DocValueFormat docValueFormat(@Nullable String format, ZoneId timeZone) {
             return DocValueFormat.BOOLEAN;
         }
 
+        @Override
+        public Query termQuery(Object value, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            Query query = new TermQuery(new Term(name(), indexedValueForSearch(value)));
+            if (boost() != 1f) {
+                query = new BoostQuery(query, boost());
+            }
+            if (isSearchable() && hasDocValues()) {
+                return new IndexOrDocValuesQuery(
+                    query,
+                    SortedNumericDocValuesField.newSlowExactQuery(name(), Values.TRUE.bytesEquals(indexedValueForSearch(value)) ? 1 : 0)
+                );
+            }
+            if (hasDocValues()) {
+                return SortedNumericDocValuesField.newSlowExactQuery(name(), Values.TRUE.bytesEquals(indexedValueForSearch(value)) ? 1 : 0);
+            }
```

Comment:
We should do this early in the method (if (!isSearchable() && hasDocValues())) to not unnecessarily create TermQuery/BoostQuery objects?

Reply:
I pushed it to the beginning so we can exit early